docs/modeling/relationships.rst

```diff
@@ -0,0 +1,236 @@
+
+.. literalinclude:: configuring/sample/EFModeling.Configuring.DataAnnotations/Samples/RelationshipInverseProperty.cs
+        :language: c#
+        :lines: 13-37
+        :emphasize-lines: 20,23
+        :linenos:
+
+Fluent API
+----------
+
+To configure a relationship in the Fluent API, you start by identifying the navigation properties that make up the relationship. ``HasOne`` or ``HasMany`` identifies the navigation property on the entity type you are beginning the configuration on. You then chain a call to ``WithOne`` or ``WithMany`` to identify the inverse navigation. ``HasOne``/``WithOne`` are used for reference navigation properties and ``HasMany``/``WithMany`` are used for collection navigation properties.
+
+.. literalinclude:: configuring/sample/EFModeling.Configuring.FluentAPI/Samples/RelationshipNoForeignKey.cs
+        :language: c#
+        :lines: 6-34
+        :emphasize-lines: 8-10
+        :linenos:
+
+Single Navigation Property
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you only have one navigation property then there are parameterless overloads of ``WithOne`` and ``WithMany``. This indicates that there is conceptually a reference or collection on the other end of the relationship, but there is no navigation property included in the entity class.
+
+.. literalinclude:: configuring/sample/EFModeling.Configuring.FluentAPI/Samples/RelationshipOneNavigation.cs
+        :language: c#
+        :lines: 6-32
+        :emphasize-lines: 10
+        :linenos:
+
+Foreign Key
+^^^^^^^^^^^
+
+You can use the Fluent API to configure which property should be used as the foreign key property for a given relationship.
+
+.. literalinclude:: configuring/sample/EFModeling.Configuring.FluentAPI/Samples/RelationshipForeignKey.cs
+        :language: c#
+        :lines: 6-36
+        :emphasize-lines: 11
+        :linenos:
+
+The following code listing shows how to configure a composite foreign key.
+
+.. literalinclude:: configuring/sample/EFModeling.Configuring.FluentAPI/Samples/RelationshipCompositeForeignKey.cs
+        :language: c#
+        :lines: 7-42
+        :emphasize-lines: 13
+        :linenos:
+
+Principal Key
+^^^^^^^^^^^^^
+
+If you want the foreign key to reference a property other than the primary key, you can use the Fluent API to configure the principal key property for the relationship. The property that you configure as the principal key will automatically be setup as an alternate key (see :doc:`alternate-keys` for more information).
+
+.. literalinclude:: configuring/sample/EFModeling.Configuring.FluentAPI/Samples/RelationshipPrincipalKey.cs
+        :language: c#
+        :lines: 7-39
+        :emphasize-lines: 11
+        :linenos:
+
+The following code listing shows how to configure a composite principal key.
```

Comment:
It might be worth mentioning in a caution that the order of the properties needs to match the order of the foreign key properties.